tests/v2/validation/provisioning/import/import_cluster_test.go

```diff
@@ -0,0 +1,183 @@
+	r.session = testSession
+	r.provisioningConfig = new(provisioninginput.Config)
+	config.LoadConfig(provisioninginput.ConfigurationFileKey, r.provisioningConfig)
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(r.T(), err)
+	r.client = client
+
+	enabled := true
+	var testuser = namegen.AppendRandomString("testuser-")
+	var testpassword = password.GenerateUserPassword("testpass-")
+	user := &management.User{
+		Username: testuser,
+		Password: testpassword,
+		Name:     testuser,
+		Enabled:  &enabled,
+	}
+
+	newUser, err := users.CreateUserWithRole(client, user, "user")
+	require.NoError(r.T(), err)
+
+	newUser.Password = user.Password
+
+	standardUserClient, err := client.AsUser(newUser)
+	require.NoError(r.T(), err)
+
+	r.standardUserClient = standardUserClient
+
+	corralConfig := corral.Configurations()
+	err = corral.SetupCorralConfig(corralConfig.CorralConfigVars, corralConfig.CorralConfigUser, corralConfig.CorralSSHPath)
+	require.NoError(r.T(), err, "error reading corral configs")
+}
+
+func (r *ImportProvisioningTestSuite) TestProvisioningImportCluster() {
+	corralPackage := corral.PackagesConfig()
+
+	if corralPackage.HasCustomRepo != "" {
+		err := corral.SetCustomRepo(corralPackage.HasCustomRepo)
+		require.Nil(r.T(), err, "error setting remote repo")
+	}
+	newPackages := []string{}
+	if len(corralPackage.CorralPackageImages) == 0 {
+		r.T().Error("No Corral Packages to Test")
+	}
+
+	_, ok := corralPackage.CorralPackageImages[k3sImportPackageName]
+	require.True(r.T(), ok, "Please ensure %s package is in corralPackageImages", k3sImportPackageName)
+
+	var importPackageName string
+
+	for packageName, packageImage := range corralPackage.CorralPackageImages {
+		newPackageName := namegen.AppendRandomString(packageName)
+		if packageName == k3sImportPackageName {
+			importPackageName = newPackageName
+		}
+
+		newPackages = append(newPackages, newPackageName)
+
+		corralRun, err := corral.CreateCorral(r.session, newPackageName, packageImage, corralPackage.HasDebug, corralPackage.HasDebug)
+		require.NoError(r.T(), err, "error creating corral %v", packageName)
+
+		r.T().Logf("Corral %v created successfully", packageName)
+		require.NotNil(r.T(), corralRun, "corral run had no restConfig")
+	}
+
+	importCluster := apiv1.Cluster{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      namegen.AppendRandomString("auto-import"),
+			Namespace: "fleet-default",
+		},
+	}
+
+	decodedKubeConfig, err := corral.GetKubeConfig(importPackageName)
+	require.NoError(r.T(), err)
+
+	config, err := clientcmd.RESTConfigFromKubeConfig(decodedKubeConfig)
+	require.NoError(r.T(), err)
+
+	_, err = clusters.CreateK3SRKE2Cluster(r.client, &importCluster)
+	require.NoError(r.T(), err)
+
+	backoff := wait.Backoff{
+		Duration: 5 * time.Second,
+		Factor:   1.1,
+		Jitter:   0.1,
+		Steps:    10,
+	}
+
+	updatedCluster := new(apiv1.Cluster)
+
+	err = wait.ExponentialBackoff(backoff, func() (finished bool, err error) {
```

Comment:
Can you elaborate on the need for this? This more so me trying to understand as opposed to calling out an issue.

Reply:
its just another type of wait as far as I'm concerned. It basically just waits longer between each time the internal func() has finished=false